hub3/ead/dao.go

```diff
@@ -0,0 +1,622 @@
+	t(s, "inventoryTitle", cfg.InventoryTitle, rdf.NewLiteral)
+
+	return triples
+}
+
+func (cfg *DaoConfig) fragmentGraph(file *eadpb.File) (*fragments.FragmentGraph, error) {
+	subjectBase := fmt.Sprintf("%s/%s/archive/%s/%s", config.Config.RDF.BaseURL, cfg.OrgID, cfg.ArchiveID, cfg.InventoryID)
+	id := fmt.Sprintf("%s-%s", cfg.InventoryID, file.Filename)
+	header := createHeader(cfg, id, subjectBase, "mets")
+	rm := fragments.NewEmptyResourceMap()
+
+	for idx, t := range cfg.fileTriples(header.EntryURI, file) {
+		if err := rm.AppendOrderedTriple(t, false, idx); err != nil {
+			return nil, err
+		}
+	}
+
+	fg := fragments.NewFragmentGraph()
+	fg.Meta = header
+	fg.Tree = &fragments.Tree{
+		CLevel:      cfg.InventoryPath,
+		UnitID:      cfg.InventoryID,
+		MimeTypes:   []string{file.MimeType},
+		SortKey:     uint64(file.SortKey),
+		Title:       cfg.InventoryTitle,
+		InventoryID: cfg.ArchiveID,
+		Label:       file.Filename,
+	}
+
+	b, err := proto.Marshal(file)
+	if err != nil {
+		return fg, fmt.Errorf("unable to marshal protobuf message: %#v", err)
+	}
+
+	fg.ProtoBuf = &fragments.ProtoBuf{
+		MessageType: "eadpb.File",
+		Data:        fmt.Sprintf("%x", b),
+	}
+	fg.SetResources(rm)
+
+	return fg, nil
+}
+
+func (cfg *DaoConfig) findingAidFragmentGraph(fa *eadpb.FindingAid) (*fragments.FragmentGraph, error) {
+	// remove files because we don't want them to be stored
+	fa.Files = []*eadpb.File{}
+
+	subjectBase := fmt.Sprintf("%s/%s/archive/%s/%s", config.Config.RDF.BaseURL, cfg.OrgID, fa.ArchiveID, fa.InventoryID)
+	id := fmt.Sprintf("%s-findingaid", fa.GetInventoryID())
+	header := createHeader(cfg, id, subjectBase, "findingaid")
+
+	rm := fragments.NewEmptyResourceMap()
+
+	for idx, t := range findingAidTriples(header.EntryURI, fa) {
+		if err := rm.AppendOrderedTriple(t, false, idx); err != nil {
+			return nil, err
+		}
+	}
+
+	fg := fragments.NewFragmentGraph()
+	fg.Meta = header
+	fg.Tree = &fragments.Tree{
+		CLevel:      fa.InventoryPath,
+		UnitID:      fa.InventoryID,
+		Title:       fa.InventoryTitle,
+		InventoryID: fa.ArchiveID,
+		Label:       "findingaid",
+	}
+
+	b, err := proto.Marshal(fa)
+	if err != nil {
+		return fg, fmt.Errorf("unable to marshal protobuf message: %#v", err)
+	}
+
+	fg.ProtoBuf = &fragments.ProtoBuf{
+		MessageType: "eadpb.FindingAid",
+		Data:        fmt.Sprintf("%x", b),
+	}
+
+	fg.SetResources(rm)
+
+	return fg, nil
+}
+
+func getMetsFilePath(archiveID, uuid string) string {
+	return path.Join(config.Config.EAD.CacheDir, archiveID, "mets", uuid)
+}
+
+func getDaoConfigPath(archiveID, uuid string) string {
+	return getMetsFilePath(archiveID, uuid) + ".json"
+}
+
+func getMetsDirPath(archiveID string) string {
+	return path.Join(config.Config.EAD.CacheDir, archiveID, "mets")
+}
+
+func GetDaoConfig(archiveID, uuid string) (DaoConfig, error) {
+	var cfg DaoConfig
+
+	daoPath := getDaoConfigPath(archiveID, uuid)
+	if _, err := os.Stat(daoPath); os.IsNotExist(err) {
```

Comment:
What should happen if there is an error but it is not a FileNotFoundError?

Reply:
Yes, it is either there or not.